### PR TITLE
Fix bean validation for type Optional

### PR DESCRIPTION
**Pull request by MelleD**
#8690

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee]
@bbdouglas
@sreeshas 
@jfiala 
@lukoyanov 
@cbornet 
@jeff9finger 
@karismann 
@Zomzog 
@lwlee2608 
@nmuesch

### CODE REVIEW

**Comment by MelleD** [comment]
The current case is not tested. Can someone give me a tip where and how I extend a test case?
I need a PathVariable and RequestParam which are not required

**Comment by MelleD** [comment]
Hey @bbdouglas
@sreeshas
@jfiala
@lukoyanov
@cbornet
@jeff9finger
@karismann
@Zomzog
@lwlee2608
@nmuesch

Anything left to do or ready to merge?

File: modules/openapi-generator/src/main/resources/JavaSpring/beanValidationPathParams.mustache [hunk 1]
```diff
@@ -1 +1 @@
-{{! PathParam is always required, no @NotNull necessary }}{{>beanValidationCore}}
\ No newline at end of file
+{{! PathParam is always required, no @NotNull necessary }}{{^useOptional}}{{>beanValidationCore}}{{/useOptional}}{{#useOptional}}{{#required}}{{>beanValidationCore}}{{/required}}{{/useOptional}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaSpring/beanValidationPathParams.mustache:1]: path parameter is always required. Does it need the `{{#required}} ... {{/required}}` check?

Comment (MelleD) [reply modules/openapi-generator/src/main/resources/JavaSpring/beanValidationPathParams.mustache]: Good catch thanks. Revert it

**Comment by MelleD** [comment]
@wing328 any update here?

**Comment by MelleD** [comment]
@wing328 Why is PR open so long? Are there any hints?

**Comment by wing328** [comment]
LGTM but I've not tested it locally. Sorry for the delay in review and merge.

**Pull request merged**